packages/client-config/src/client-config-writer/client_config_writer.ts

```diff
@@ -7,6 +7,7 @@ import {
   ClientConfigVersion,
 } from '../client-config-types/client_config.js';
 import { ClientConfigFormatter } from './client_config_formatter.js';
+import { FilesWrittenResult } from '@aws-amplify/plugin-types';
```

Comment:
ok. I noticed that this type is in plugin types.

I think it's ok to have 3 version of this type under different names that fit local package naming.

Reply:
We could also have this type live here and then define specific method return types as `type GenerateClientConfigToFileResult = FilesWrittenResult`.
Then we can add to specific method return values without having to modify the shared part of the type.

Reply:
I'd refrain from promoting it to plugin types.
I think coupling does more harm here than duplication. It's a simple type.
De-duplication can follow if needed later.

Reply:
I'll split this up to local types and remove the change in `plugin-types`